tests/Unit/Domain/BoundaryConditions/Test_BoundaryCondition.cpp

```diff
@@ -0,0 +1,51 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <charm++.h>
+#include <memory>
+#include <pup.h>
+#include <string>
+
+#include "Domain/BoundaryConditions/BoundaryCondition.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Parallel/CharmPupable.hpp"
+
+namespace {
+class BoundaryCondition
+    : public domain::boundary_conditions::BoundaryCondition {
+ public:
+  BoundaryCondition() = default;
+
+  std::unique_ptr<domain::boundary_conditions::BoundaryCondition> get_clone()
+      const noexcept override {
+    return std::make_unique<BoundaryCondition>(*this);
+  }
+
+  explicit BoundaryCondition(CkMigrateMessage* msg) noexcept
+      : domain::boundary_conditions::BoundaryCondition(msg) {}
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+  WRAPPED_PUPable_decl_base_template(
+      domain::boundary_conditions::BoundaryCondition, BoundaryCondition);
+#pragma GCC diagnostic pop
+
+  void pup(PUP::er& p) override {
+    domain::boundary_conditions::BoundaryCondition::pup(p);
```

Comment:
(optional) Could be nicer to define `private: using Base = ...` at the top and use the type alias throughout this class.